scripts/average_checkpoints.py

```diff
@@ -44,20 +46,18 @@ def average_checkpoints(inputs):
             )
 
         for k in params_keys:
-            if k not in params_dict:
-                params_dict[k] = []
             p = model_params[k]
             if isinstance(p, torch.HalfTensor):
                 p = p.float()
-            params_dict[k].append(p)
+            if k not in params_dict:
+                params_dict[k] = p
+            else:
+                params_dict[k] += p
```

Comment:
We need to be careful here with shared parameters, I believe right now it is broken if you have shared encoder/decoder input embeddings (encoder.embed_tokens.weight and decoder.embed_tokens.weight) as they get updated several times

Reply:
Thanks for pointing out this. Could you kindly give me an example of shared parameters so that I can better unittest this code ?

Reply:
See pr #747